### PR TITLE
Added a new datasource for fetching the available alloydb locations

### PR DESCRIPTION
**Pull request by ravisiddhu**
This PR is for https://github.com/hashicorp/terraform-provider-google/issues/13812

A new datasource `google_alloydb_locations` for fetching the details of available alloydb locations. It receives `project`  as arguments and returns the details of all the available alloydb locations.

I have added documentation and tested this new data source with unit tests.

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [X] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

```release-note:new-datasource
`google_alloydb_locations`
```

### CODE REVIEW

**Review by NickElliot — CHANGES_REQUESTED** [review]

File: mmv1/third_party/terraform/website/docs/d/alloydb_locations.html.markdown [hunk 1]
```diff
@@ -0,0 +1,41 @@
+---
+subcategory: "Alloydb"
+description: |-
+  Fetches the details of available locations.
+---
+
+# google\_alloydb\_locations
+
+Use this data source to get information about the available locations.
```
[the remark this answers is not in the document]
Comment (ravisiddhu) [reply mmv1/third_party/terraform/website/docs/d/alloydb_locations.html.markdown]: Done.

**Comment by NickElliot** [comment]
LGTM! @zli82016 for final review

File: mmv1/third_party/terraform/data_sources/data_source_alloydb_locations.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+	}
+	billingProject = project
+
+	// err == nil indicates that the billing_project value was found
+	if bp, err := getBillingProject(d, config); err == nil {
+		billingProject = bp
+	}
+
+	url, err := ReplaceVars(d, config, "{{AlloydbBasePath}}projects/{{project}}/locations")
+	log.Printf("[ERROR] kanthara url : %s", url)
```
Comment (zli82016) [inline mmv1/third_party/terraform/data_sources/data_source_alloydb_locations.go:85]: Is this log required?

**Comment by zli82016** [comment]
Overall LTGM.

**Review by zli82016 — APPROVED** [review]

**Review by NickElliot — APPROVED** [review]

File: mmv1/third_party/terraform/data_sources/data_source_alloydb_locations.go (hunk 2 above)
Comment (ravisiddhu) [reply mmv1/third_party/terraform/data_sources/data_source_alloydb_locations.go]: It is not needed, I have removed it.

**Pull request merged**